scheduler/cloudformation/cloudformation.go

```diff
@@ -268,11 +271,129 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 		if err := <-done; err != nil {
 			return err
 		}
+		done = make(chan error, 1)
+		s.waitForDeploymentToStabilize(ctx, stackName, ss, done)
```

Comment:
This has the potential to produce incorrect results, because DescribeStack is called while this deployment isn't holding the stack operation lock. Maybe we should make `performStackOperation` return a `DescribeStackOutput`.